rt/test/TESTOPR2.MLC

```diff
@@ -0,0 +1,1443 @@
+         AIF   ('&X2D' NE '+241').X2ERR
+&X2D     SETC  X2D('7FFFFFFF')
+         MNOTE 'X2D(''7FFFFFFF'')=&X2D'
+         AIF   ('&X2D' NE '+2147483647').X2ERR
+&X2D     SETC  X2D('FFFFFFF1')
+         MNOTE 'X2D(''FFFFFFF1'')=&X2D'
+         AIF   ('&X2D' NE '-15').X2ERR
+*----------------------------------------------------------------------
+*        z390 tests
+
+&X2D     SETC  X2D('F1')
+         MNOTE 'X2D(''F1'')=&X2D'
+         AIF   ('&X2D' NE '+241').X2ERR
+*----------------------------------------------------------------------
+         MNOTE '********** End   X2D tests **********'
+*======================================================================
+.SKPX2Y ANOP
+***********************************************************************
+*        MBOX  'Test DCLEN, DCVAL, DEQUOTE'
+         MBOX  'Test DCLEN'
+*AGO .SKPDC
+*======================================================================
+         MNOTE '********** Begin DCLEN tests **********'
+*----------------------------------------------------------------------
+*        HLASM LR examples
+
+&DCLEN   SETA  DCLEN('')                 null string
+         MNOTE 'DCLEN('''')=&DCLEN'
+         AIF   (&DCLEN NE 0).DCLERR
+&DCLEN   SETA  DCLEN('''')               single apostrophe
+         MNOTE 'DCLEN('''''''')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('''''')             two apostrophes
+         MNOTE 'DCLEN('''''''''''')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('&&')               two ampersands
+         MNOTE 'DCLEN(''&&&&'')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('a''''b')           string is "a'b"
+         MNOTE 'DCLEN(''a''''''''b'')=&DCLEN'
+         AIF   (&DCLEN NE 3).DCLERR
+&DCLEN   SETA  DCLEN('a''''b&&c')        string is "a'b&c""
+         MNOTE 'DCLEN(''a''''''''b&&&&c'')=&DCLEN'
+         AIF   (&DCLEN NE 5).DCLERR
+&DCLEN   SETA  DCLEN('&&&&'.'''''''')    string is "&&''"
+         MNOTE 'DCLEN(''&&&&&&&&''.'''''''''''''''')=&DCLEN'
+         AIF   (&DCLEN NE 4).DCLERR
+*----------------------------------------------------------------------
+*        z390 tests
+
+&DCLSTR  SETC  '''''''''&&&&&&&&1'''''''''  len 17; see MNOTE below
+&K       SETA  K'&DCLSTR 
+&DDCLSTR SETC  DOUBLE('&DCLSTR')            double for MNOTE and DC
+*
+         MNOTE 'K''&&DCLSTR=&K  &&DCLSTR = >&DDCLSTR<'
+***********************************************************************
+*        On z390 the preceding MNOTE displays
+*
+*  MNOTE 'K'&DCLSTR=17  &DCLSTR = >''''''''&&&&&&&&&&&&&&&&1''''''''<'
+*
+*        Using HLASM the MNOTE displays
+*
+* +K'&DCLSTR=17  &DCLSTR = >''''&&&&&&&&1''''<           
+*
+*        Looks like mz390 just shows the MNOTE with the message
+*        text doubled, not the actual output of the MNOTE.
+*
+*        The following DC statement is the same for both z390 & HLASM.
+***********************************************************************
+         PUSH  PRINT
```

Comment:
I presume this indicates a shortcoming of z390 MNOTE processing. If so, then an issue should be created to cover the case.